src/datasets/load.py

```diff
@@ -1574,46 +1584,64 @@ def dataset_module_factory(
     # Try remotely
     elif is_relative_path(path) and path.count("/") <= 1:
         try:
-            _raise_if_offline_mode_is_enabled()
-            hf_api = HfApi(config.HF_ENDPOINT)
+            # Get the Dataset Card + get the revision + check authentication all at in one call
+            # We fix the commit_hash in case there are new commits in the meantime
+            api = HfApi(
+                endpoint=config.HF_ENDPOINT,
+                token=download_config.token,
+                library_name="datasets",
+                library_version=__version__,
+                user_agent=get_datasets_user_agent(download_config.user_agent),
+            )
             try:
-                dataset_info = hf_api.dataset_info(
+                _raise_if_offline_mode_is_enabled()
+                dataset_readme_path = api.hf_hub_download(
                     repo_id=path,
+                    filename=config.REPOCARD_FILENAME,
+                    repo_type="dataset",
                     revision=revision,
-                    token=download_config.token,
-                    timeout=100.0,
+                    proxies=download_config.proxies,
                 )
+                commit_hash = os.path.basename(os.path.dirname(dataset_readme_path))
+            except EntryNotFoundError as e:
+                if "internet connection" in str(e).lower():
+                    raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({e.__class__.__name__})") from e
+                commit_hash = api.dataset_info(
+                    path,
+                    revision=revision,
+                    timeout=100.0,
+                ).sha
             except (
                 OfflineModeIsEnabled,
                 requests.exceptions.ConnectTimeout,
                 requests.exceptions.ConnectionError,
             ) as e:
                 raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({e.__class__.__name__})") from e
+            except GatedRepoError as e:
+                message = f"Dataset '{path}' is a gated dataset on the Hub."
+                if "401 Client Error" in str(e):
+                    message += " You must be authenticated to access it."
+                elif "403 Client Error" in str(e):
+                    message += f" Visit the dataset page at https://huggingface.co/datasets/{path} to ask for access."
```

Comment:
You should be able to check `gated_error.response.status_code` (or something like this) to check the status code.

Reply:
cool, done !